### PR TITLE
Javascript Carousel First Draft

### PR DESCRIPTION
**Pull request by neldaza**
https://user-images.githubusercontent.com/88061673/140233352-753c1fa3-7737-4125-b468-8a5f45286258.mov

### CODE REVIEW

File: javascript-carousel/main.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+$circleThree.addEventListener('click', pictureThree);
+
+function pictureFour(event) {
+  $img.setAttribute('src', 'images/004.png');
+  $circleOne.className = 'circle circle-one';
+  $circleTwo.className = 'circle circle-two';
+  $circleThree.className = 'circle circle-three';
+  $circleFour.className = 'circle circle-four circle-background';
+  $circleFive.className = 'circle circle-five';
+}
```
Comment (codingcodymiller) [inline javascript-carousel/main.js:50]: This is all very manual. If you want to add additional images, you need to make a new function every time. Instead, I recommend you make an array of all the possible `src` values for the image in the carousel, and then make a variable to keep track of the index for the current image. You should also make use of a loop over the dots to show the one for the current image as solid, and the rest as hollow. I suggest putting a `data-` attribute on each dot icon, maybe something like `data-image-index`

**Review by codingcodymiller — CHANGES_REQUESTED** [review]
Nice work so far! One note to get you started:

File: javascript-carousel/main.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  $img.setAttribute('data-image-index', parseInt($img.getAttribute('data-image-index')) - 1);
+  for (let i = 0; i < $circleSelectorAll.length; i++) {
+    if (parseInt($circleSelectorAll[i].getAttribute('data-image-index')) === parseInt($img.getAttribute('data-image-index'))) {
+      $circleSelectorAll[i].className = 'circle circle-background';
+    } else {
+      $circleSelectorAll[i].className = 'circle';
+    }
+  }
+  timerId = setInterval(updatingPicture, 3000);
+}
```
Comment (thebearingedge) [inline javascript-carousel/main.js:69]: This is pretty much the same code twice. If you update the current image and just roll with it, you don't need all of this code.

File: javascript-carousel/main.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  currentImage = parseInt($img.getAttribute('data-image-index')) + 1;
+  $img.setAttribute('src', imageSrcArray[currentImage]);
+  $img.setAttribute('data-image-index', parseInt($img.getAttribute('data-image-index')) + 1);
+  for (let i = 0; i < $circleSelectorAll.length; i++) {
+    if (parseInt($circleSelectorAll[i].getAttribute('data-image-index')) === parseInt($img.getAttribute('data-image-index'))) {
+      $circleSelectorAll[i].className = 'circle circle-background';
+    } else {
+      $circleSelectorAll[i].className = 'circle';
+    }
+  }
```
Comment (thebearingedge) [inline javascript-carousel/main.js:96]: This is pretty much the same code twice. If you update the current image and just roll with it, you don't need all of this code.

**Review by thebearingedge — CHANGES_REQUESTED** [review]

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**